assets/js/components/notifications/dashboard-auth-alert.js

```diff
@@ -24,22 +24,23 @@ import { __, sprintf } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
+import { getModulesData } from '../../util';
 import Notification from '../notifications/notification';
 
 const DashboardAuthAlert = () => {
 	const { admin: { connectURL } } = global.googlesitekit;
-	const { currentAdminPage } = global.googlesitekit.admin;
-	const product = currentAdminPage
-		.replace( /googlesitekit|module|-/g, ' ' )
-		.replace( /(^\w{1})|(\s{1}\w{1})/g, ( match ) => match.toUpperCase() )
-		.trim();
+	const { requiredScopes, grantedScopes } = global.googlesitekit.setup;
+
+	const productUrl = requiredScopes.filter( ( e ) => ! grantedScopes.includes( e ) )[ 0 ];
+	const productSlug = productUrl ? productUrl.toString().split( '/' ).pop() : '';
+	const product = getModulesData()[ productSlug ] ? getModulesData()[ productSlug ].name : '';
```

Comment:
This isn't sufficient to get the slug of the service/module as many scopes have `.readonly` or other variants tagged on the end. Let's use a regex to match `https://www.googleapis.com/auth/{THIS}.whatevermaybe`.

We should keep the page-based logic as a fallback as we should never use an empty string for `product` here.
Let's rename `product` to `productName` as well 👍

Reply:
Updated